x-pack/plugins/observability_solution/slo/server/services/historical_summary_client.ts

```diff
@@ -63,8 +63,8 @@ export class DefaultHistoricalSummaryClient implements HistoricalSummaryClient {
 
   async fetch(params: FetchHistoricalSummaryParams): Promise<HistoricalSummaryResponse> {
     const dateRangeBySlo = params.list.reduce<Record<SLOId, DateRange>>(
-      (acc, { sloId, timeWindow }) => {
-        acc[sloId] = getDateRange(timeWindow);
+      (acc, { sloId, timeWindow, range }) => {
+        acc[sloId] = range ?? getDateRange(timeWindow);
```

Comment:
@shahzad31 We should also take into consideration if it is a rolling or a calendar-defined SLO. For calendar-aligned it should get the daterange of the defined timeWindow.

Reply:
Now that I am thinking about it, if you completely hide the datepicker for calendar-aligned, then this code is fine